src/libertem/web/server.py

```diff
@@ -100,7 +101,30 @@ def make_app(event_registry, shared_state):
             "state": shared_state,
             "event_registry": event_registry,
         }),
-    ], **settings)
+        (r"/api/session/", SessionHandler, {
+            "state": shared_state,
+            "event_registry": event_registry,
+        }),
+        (r"/api/session/([^/][0-9a-f-]+)/datasets/([^/][0-9a-f]+)/", SessionDatasetHandler, {
+            "state": shared_state,
+            "event_registry": event_registry,
+        }),
+    ]
+
+    shared_state.instance_type = instance_type
+    if instance_type == "restricted":
+        loop = asyncio.get_event_loop()
+        loop.run_until_complete(shared_state.set_local_executor(instance_config))
```

Comment:
This is where I set the config from the file in restricted mode. So the user does not get to choose how much resources he can use. That's done by the server admin.